pandas/core/util/hashing.py

```diff
@@ -264,7 +287,7 @@ def hash_array(vals, encoding='utf8', hash_key=None, categorize=True):
 
         try:
             vals = hashing.hash_object_array(vals, hash_key, encoding)
-        except TypeError:
+        except (TypeError, ValueError):
```

Comment:
the hashing doesn't handle fixed-length arrays (it prob could, but we don't generally have them). if its an object array it will work.
```
In [21]: np.array(['E'])
Out[21]: 
array(['E'], 
      dtype='<U1')

In [22]: np.array(['E'], dtype=object)
Out[22]: array(['E'], dtype=object)

In [20]: pd.core.util.hashing.hash_array(np.array(['E'], dtype=object), categorize=False)
Out[20]: array([16179053688037232491], dtype=uint64)
```